name.abuchen.portfolio/src/name/abuchen/portfolio/model/InvestmentPlan.java

```diff
@@ -481,16 +488,22 @@ private TransactionPair<?> createAccountTx(CurrencyConverter converter, LocalDat
     {
         long txAmount = amount;
 
-        AccountTransaction.Type type;
+        AccountTransaction.Type transactionType;
         if (txAmount > 0)
         {
-            type = AccountTransaction.Type.DEPOSIT;
+            transactionType = AccountTransaction.Type.DEPOSIT;
         }
         else
         {
-            type = AccountTransaction.Type.REMOVAL;
+            transactionType = AccountTransaction.Type.REMOVAL;
             txAmount = -txAmount;
         }
+        if (type == Type.INTEREST)
+        {
+            transactionType = AccountTransaction.Type.INTEREST;
+            txAmount -= taxes;
```

Comment:
The logic for all other transactions is:

the amount is the monetary amount of the transactions.
taxes are deducted from this amount implicitly.

If you remove the taxes here, then the taxes will be removed again.
And it is not consistent with the other transactions.

Reply:
Hello, interesting, this was there because otherwise transaction showed wrong value when I was testing it IIRC.

Reply:
![obrazek](https://github.com/portfolio-performance/portfolio/assets/20244554/f2a12f3f-5959-41d3-a897-1b40c962f917)
@buchen Created transaction is adding taxes to the amount, not subtracting it.

Reply:
<img width="1102" alt="Bildschirmfoto 2024-06-08 um 14 24 40" src="https://github.com/portfolio-performance/portfolio/assets/587976/d7850070-13de-4a62-98a2-b704b6a907d7">

<img width="1147" alt="Bildschirmfoto 2024-06-08 um 14 24 59" src="https://github.com/portfolio-performance/portfolio/assets/587976/d29d79ad-3976-4f0f-b69e-d999337b31e3">

<img width="432" alt="Bildschirmfoto 2024-06-08 um 14 30 35" src="https://github.com/portfolio-performance/portfolio/assets/587976/3d1cd6ab-7330-41cd-a451-b416c60b7b5b">


I believe, if you have a look at the cash account, you should see that 100 is added to the balance.

The view says:

* You got 120 of interest payments
* You had to pay 20 of taxes
* Your payout was 100

Is that also what you see?

That is the same behavior that is implemented for the investment plan of the purchase (only there the reverse logic applies because one pays an amount X plus the fees to purchase the instrument).

Now I agree that entering the values might be counterintuitive to the user. Do you think that adding gross and net values (and making sure both are in sync) is more helpful?

Reply:
Yes, I see that.
![image](https://github.com/portfolio-performance/portfolio/assets/20244554/20cb3559-0e96-4b9c-8d2a-84ce4b09fb14)

I think problem is in consistency - if I create Interest transaction from Cash Account and from Plan I would need to insert different values. If I insert same values (as user assuming using gross everywhere and taxes) the result transactions are different:

Adding Interest through Account (100 as gross, payout 80):
![image](https://github.com/portfolio-performance/portfolio/assets/20244554/b7a4fcde-a9bd-4fe0-b788-8fe5bf6eaeb7)

Adding Interest through Plan (also 100 assuming it's gross, but it's actually net now, resulting in 120 gross as in first screenshot)
![image](https://github.com/portfolio-performance/portfolio/assets/20244554/41192c24-9d98-4a06-8f5e-199d0a58d7e8)


Result transactions with different net value payouts:
![image](https://github.com/portfolio-performance/portfolio/assets/20244554/2be652eb-5485-48c2-ae64-9e7f848649d5)

Reply:
I think the best way is to have the very same input controls (gross value, taxes, net value) when creating the investment plan as when creating a single transaction. I can do this.